plutus-core/cost-model/create-cost-model/CreateBuiltinCostModel.hs

```diff
@@ -130,860 +117,263 @@ builtinCostModelNames = BuiltinCostModelBase
-  -- x - y
-  let memModel = ModelTwoArgumentsSubtractedSizes $ ModelSubtractedSizes 0 1 1
-  pure $ CostingFun cpuModel memModel
-
-remainderInteger :: MonadR m => SomeSEXP (Region m) -> m (CostingFun ModelTwoArguments)
-remainderInteger = quotientInteger
-
-modInteger :: MonadR m => SomeSEXP (Region m) -> m (CostingFun ModelTwoArguments)
-modInteger = divideInteger
-
--- FIXME: should probably be piecewise (harmless, but may overprice some comparisons a bit)
-equalsInteger :: MonadR m => SomeSEXP (Region m) -> m (CostingFun ModelTwoArguments)
-equalsInteger cpuModelR = do
-  cpuModel <- ModelTwoArgumentsMinSize <$> readModelMinSize cpuModelR
-  pure $ CostingFun cpuModel boolMemModel
-
-lessThanInteger :: MonadR m => SomeSEXP (Region m) -> m (CostingFun ModelTwoArguments)
-lessThanInteger cpuModelR = do
-  cpuModel <- ModelTwoArgumentsMinSize <$> readModelMinSize cpuModelR
-  pure $ CostingFun (cpuModel) boolMemModel
-
-lessThanEqualsInteger :: MonadR m => SomeSEXP (Region m) -> m (CostingFun ModelTwoArguments)
-lessThanEqualsInteger cpuModelR = do
-  cpuModel <- ModelTwoArgumentsMinSize <$> readModelMinSize cpuModelR
-  pure $ CostingFun cpuModel boolMemModel
-
-
----------------- Bytestrings ----------------
-
-appendByteString :: MonadR m => SomeSEXP (Region m) -> m (CostingFun ModelTwoArguments)
-appendByteString cpuModelR = do
-  cpuModel <- ModelTwoArgumentsAddedSizes <$> readModelAddedSizes cpuModelR
-  let memModel = ModelTwoArgumentsAddedSizes $ ModelAddedSizes 0 1
-  pure $ CostingFun cpuModel memModel
-
-consByteString ::  MonadR m => SomeSEXP (Region m) -> m (CostingFun ModelTwoArguments)
-consByteString cpuModelR = do
-  m <- readModelLinearInY cpuModelR
-  let cpuModel = ModelTwoArgumentsLinearInY m
-      memModel = ModelTwoArgumentsAddedSizes $ ModelAddedSizes 0 1
-  pure $ CostingFun cpuModel memModel
-
-
-{- | Return a substring of a bytestring with a specified start point and length.
-   Plutus Core bytestrings are implemented using Data.ByteString, which
-   represents a (strict) bytestring as a C array of bytes together with a
-   pointer into that and a length.  The sliceByteString function is implemented
-   using 'take' and 'drop', and these work by modifying the pointer and length:
-   no bytes are copied so sliceByteString requires constant time and a constant
-   memory overhead.  There's a mismatch here because the Haskell model which we
-   defined for SliceByteString is linear in the length of the bytestring;
-   however we can still use that to model the constant cost inferred in the R
-   code.
+  cpuModels :: BuiltinCostModelBase (Const (SomeSEXP s)) <- costModelsR bmfile rfile
+  let getParams
+        :: (SomeSEXP s -> R s model)
+        -> (forall f. BuiltinCostModelBase f -> f model)
+        -> R s (CostingFun model)
+      getParams readCF param = do
+        let memModel = getId $ param builtinMemoryModels
+        cpuModel <- readCF $ getConst $ param cpuModels
```

Comment:
That's the way it's always been, and that's been OK because memory models don't change too much. Also we're not  doing any analysis to infer the memory models, just making them up based on what we know about the implemenations of the builtins.  What we could do is have a separate (versioned) JSON file for the memory models and read them from there: you'd have to write the JSON by hand though, which might be a bit fragile.  Maybe we could use some other format (which is essentially what we have in the Haskell now anyway). Actually maybe it's better just to keep them in `buikltinCostModel.json` anyway since if we eventually have mulitple versions of that it'd be easier to manage things if the memory models are in the same place as the CPU models.